Bash code for fizzbuzz program:

PavithraUmakanthan opened the pull request:
This program finds the numbers divisbile by 3, 5 and both. Replace them with the words 'fizz', 'buzz' and 'fizzbuzz' accordingly.

This program loops the numbers from 1 to 100 only.

Commit:
Bash code for fizzbuzz program:

This program finds the numbers divisbile by 3, 5 and both. Replace them with the words 'fizz', 'buzz' and 'fizzbuzz' accordingly.

This program loops the numbers from 1 to 100 only.

Commit:
Added shebang and double quotes for echo.

dougireton commented on fizzbuzz.sh:
```diff
@@ -26,4 +27,23 @@
 
+function second(){
+   if (( $1 % 3 == 0 )) && (( $1 % 5 == 0 ));then
+       var="fizzbuzz"
```
Variables in Bash are global by default, which is global mutable state. Please make all variables `local`.

See the accepted answer here to `echo` your return value to the calling function: https://stackoverflow.com/questions/17336915/return-value-in-a-bash-function

dougireton commented on fizzbuzz.sh:
```diff
@@ -1,3 +1,4 @@
+#!/bin/bash
```
Please use the more portable `#!/usr/bin/env bash` instead.

See https://kvz.io/blog/2013/11/21/bash-best-practices/

dougireton replied:
Please start all bash scripts with these lines:
```shell
set -o errexit
set -o pipefail
set -o nounset
```
https://kvz.io/blog/2013/11/21/bash-best-practices/

dougireton commented on fizzbuzz.sh:
```diff
@@ -26,4 +27,23 @@
+   fi    
+}
 
+function first(){
```
This should be the `main()` function.

dougireton commented on fizzbuzz.sh:
```diff
@@ -26,4 +27,23 @@
 #      output:fizz
 
+function second(){
+   if (( $1 % 3 == 0 )) && (( $1 % 5 == 0 ));then
```
`$1` is pretty cryptic. Try this instead:
```bash
local number=$1
if (( "$number" % 3 == 0 )) && (( "$number" % 5 == 0 )); then
```

dougireton commented on fizzbuzz.sh:
```diff
@@ -26,4 +27,23 @@
+}
 
+function first(){
+    for n in $(seq 1 100)
```
no need to call out to a shell command here. Bash has built-in ranges:

`for n in {1..100};`

dougireton commented on fizzbuzz.sh:
```diff
@@ -26,4 +27,23 @@
 #      input:21
 #      output:fizz
 
+function second(){
```
There are two ways to declare functions in Bash. If you use the `function` keyword, you don't need the parenthesis.

```bash
function foo {
    echo "foo"
}

bar () {
    echo "bar"
}
```

dougireton replied:
`second()` is a cryptic function name. Maybe try `fizzbuzz()`?

dougireton commented:
Shell scripts should have no file extension: https://google.github.io/styleguide/shell.xml?showone=File_Extensions#File_Extensions

Commit:
Renamed the file fizzbuzz.sh to fizzbuzz

Commit:
Updated the fizzbuzz code as per the Doug's feedback:

1.Added /usr/bin/env because this helps in looking for the default version of the program is in the current environment.
2.Added the exit code to exit when the command fails, when using the undeclared variables and pipeline failure
3.Updated the function names to fizzbuzz and main so that the funaction names are self-explantory of what the function is doing.
4.Updated the global variables to local variables since they are not used outside the function.
5.Updated the shell command for range (seq 1 100) to Bash range. Since the whole code is in Bash language.Updated the fizzbuzz code as per the Doug's feedback.

Commit:
1.Removed the local variable "result" in the main function since I am using echo to directly print the return from the fizzbuzz function.
2. Removed the global variable $n in the fizzbuzz function since it is confusing and replaced with $1.

dougireton reviewed (approved):
Nice work.
